flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/TableStoreSource.java

```diff
@@ -130,15 +135,29 @@ public DataStructureConverter createDataStructureConverter(
                             },
                             projectFields);
         }
-        TableStore.SourceBuilder builder =
+
+        TableStore.SourceBuilder sourceBuilder =
                 tableStore
                         .sourceBuilder()
                         .withContinuousMode(streaming)
                         .withLogSourceProvider(logSourceProvider)
                         .withProjection(projectFields)
                         .withPartitionPredicate(PredicateConverter.convert(partitionFilters))
-                        .withFieldPredicate(PredicateConverter.convert(fieldFilters));
-        return SourceProvider.of(builder.build());
+                        .withFieldPredicate(PredicateConverter.convert(fieldFilters))
+                        .withParallelism(tableStore.options().get(SCAN_PARALLELISM));
+
+        return new DataStreamScanProvider() {
+            @Override
+            public DataStream<RowData> produceDataStream(
+                    ProviderContext providerContext, StreamExecutionEnvironment env) {
+                return sourceBuilder.withEnv(env).build();
+            }
+
+            @Override
+            public boolean isBounded() {
+                return !streaming;
```

Comment:
I'm not very sure about this. Does the file store still serve as a bounded scan if it is a hybrid reading with a full scan?

Reply:
hybrid mode must be streaming.